src/DataStructures.jl

```diff
@@ -104,4 +104,6 @@ module DataStructures
     export PriorityQueue, peek
 
     include("priorityqueue.jl")
+    include("sparse_int_set.jl")
+    export SparseIntSet, cleanup!, current_id
```

Comment:
Why are we exporting `current_id` ?

```suggestion
    export SparseIntSet, cleanup!
```

Reply:
it's useful during iteration if you don't want the indices inside the packed array corresponding to the id but rather the id itself. I did it this way after experimenting with it and finding that making a tuple (id, packed_ids) during iteration impacts performance quite a lot, whereas storing it inside the iterator not at all. But yea, it's maybe not the best way?

Reply:
it has no docs, does it?

Reply:
No indeed, would you feel like keeping it and include documentation, or just leaving it out? I don't know if it's a good api

Reply:
One possibly convoluted way is to make a `IntPage` datastructure that holds the page and a counter, each time something gets pushed to the page the counter gets incremented, when popped, decremented. Then the set can check whether it's 0 to clean it up. But that will lower performance somewhat I guess since extra instructions per push and pop.